drivers/serial/uart_nrfx_uart.c

```diff
@@ -843,6 +843,10 @@ static void uart_nrfx_isr(void *arg)
 {
 	ARG_UNUSED(arg);
 
+	if (nrf_uart_event_check(uart0_addr, NRF_UART_EVENT_ERROR)) {
+		nrf_uart_event_clear(uart0_addr, NRF_UART_EVENT_ERROR);
```

Comment:
Since this is called before the callback function, it will effectively preclude reading of the error source when any interrupt is enabled. Either this should be moved after the call to `irq_callback` or the [check](https://github.com/zephyrproject-rtos/zephyr/blob/27f358de34d8db27766207cbc09a51d131344253/drivers/serial/uart_nrfx_uart.c#L256) of `ERROR` event should be removed from `uart_nrfx_err_check`. The latter option seems to be better.
Same goes for UARTE.